services/create_endpoint.go

```diff
@@ -0,0 +1,142 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/api/models"
+	"github.com/frain-dev/convoy/cache"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/frain-dev/convoy/util"
+	"github.com/oklog/ulid/v2"
+)
+
+type CreateEndpointService struct {
+	Cache        cache.Cache
+	EndpointRepo datastore.EndpointRepository
+	ProjectRepo  datastore.ProjectRepository
+
+	E         models.Endpoint
+	ProjectID string
+}
+
+func (a *CreateEndpointService) Run(ctx context.Context) (*datastore.Endpoint, error) {
+	if err := util.Validate(a.E); err != nil {
```

Comment:
@danvixent As a rule of thumb, model validations should happen in the handler.  Only other nuanced core business logic validations should happen here.

Reply:
@subomi fixed

Reply:
@subomi fixed